docusaurus/docs/introduction/plugin-types-usage

```diff
@@ -0,0 +1,133 @@
+
+:::
+
+Grafana plugin development allows for many options depending on the type of user experience you want to create. Make a plugin in one of several different types depending on whether you want to offer users a new way of visualizing data, a connection to a new database, or an integrated out-of-the-box experience.  
+
+The following table shows the plugin types currently supported by Grafana on the frontend and backend. You can also see whether or not each type is supported by the [Grafana API](https://grafana.com/api/plugins).
+
+|                |                 |                |                            |
+| -------------- | :-------------: | :------------: | :------------------------: |
+| **Type**       | **Frontend** |   **Backend**  | **SDK-supported** |
+| datasource     |       Yes       | Yes (optional) |             Yes            |
+| app            |       Yes       | Yes (optional) |             Yes            |
+| panel          |       Yes       |       No       |             Yes            |
+| renderer       |        No       |       Yes      |             No             |
+| secretsmanager |        No       |       Yes      |             No             |
+
+## Data source plugins
+
+In plugin development, you can create new data source plugins to use Grafana with new databases, similar to existing plugins such as [MongoDB](https://grafana.com/grafana/plugins/grafana-mongodb-datasource/) or [Google BigQuery](https://grafana.com/grafana/plugins/grafana-bigquery-datasource/).
+
+### Usage of data-source plugins
+
+Create and configure a Grafana data source when you want to work with data provided by a third-party service for use in Grafana Dashboards, Explore, Alerting, and so on.
+
+The data source API [(`/api/datasources`)](https://grafana.com/docs/grafana/latest/developers/http_api/data_source/) exposes endpoints to create, read, update, and delete data sources.
+
+Given any installed plugin of type `datasource`, you can create and configure any number of data sources per [Grafana organization](https://grafana.com/docs/grafana/latest/administration/organization-management/#about-organizations). After you create zero to infinity data sources, they are persisted in Grafana's database. 
+
+:::note
+
+To distinguish a Grafana data source from a data-source plugin, we sometimes refer to the latter as a _data-source instance_, that is, a configured Grafana data source with a `plugin id` type of `datasource`.
+
+:::
+
+
+### Provisioning of data sources
+
+Use the [Grafana configuration file](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/) to configure your data source plugins.
+
+Data sources can also be provisioned using Grafana’s [provisioning features](https://grafana.com/docs/grafana/latest/administration/provisioning/) so that you can use YAML files when Grafana starts up or on demand. Note that a plugin of type `datasource` must be installed before it can be provisioned.
```

Comment:
> so that you can use YAML files when Grafana starts up or on demand

This isn't the user's goal 😄  The goal is to be able to define Grafana as-code so they can adopt more of a GitOps approach (version controlled, auditable, enforceable, repeatable). Suggest reading the start of the provisioning doc for more context.

Reply:
I also think we should provide the link for [on demand](https://grafana.com/docs/grafana/latest/developers/http_api/admin/#reload-provisioning-configurations) as in the original doc

Reply:
got it, done